### PR TITLE
Test option validation in deck entrypoint

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/assign @BenTheElder 

Didn't have any validation before 😲 , working on #6575

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
Is this addressing tech debt .... or am I just swabbing the `deck`?

File: prow/cmd/deck/main.go [hunk 1]
```diff
@@ -83,36 +116,36 @@ func main() {
...
+		o.prodOnlyMain(mux)
 	}
 
 	// setup done, actually start the server
 	logrus.WithError(http.ListenAndServe(":8080", mux)).Fatal("ListenAndServe returned.")
 }
 
 // prodOnlyMain contains logic only used when running deployed, not locally
-func prodOnlyMain(mux *http.ServeMux) {
+func (o *options) prodOnlyMain(mux *http.ServeMux) {
```
Comment (BenTheElder) [inline prow/cmd/deck/main.go]: er should this be on options or take in an options?

Comment (BenTheElder) [reply prow/cmd/deck/main.go]: FWIW this whole method is a bit gross and mostly my fault.

Comment (stevekuznetsov) [reply prow/cmd/deck/main.go]: Didn't really feel strongly either way, moved it to a parameter

**Pull request merged**